src/main/java/it/ipzs/fedauthority/oidclib/OidcConfig.java

```diff
@@ -0,0 +1,448 @@
+package it.ipzs.fedauthority.oidclib;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+import it.ipzs.fedauthority.oidclib.schemas.OIDCProfile;
+
+@Configuration
+@ConfigurationProperties(prefix = "oidc")
+public class OidcConfig extends BaseConfig {
+
+	private String defaultTrustAnchor;
+	private List<String> trustAnchors = new ArrayList<>();
+	private RelyingParty relyingParty = new RelyingParty();
+	private Hosts hosts = new Hosts();
+	private List<ProviderInfo> spidProviders = new ArrayList<>();
+	private List<ProviderInfo> cieProviders = new ArrayList<>();
+	private OpenIdCredentialIssuer openidCredentialIssuer = new OpenIdCredentialIssuer();
+	private FederationEntity federationEntity = new FederationEntity();
+	private String federationTrustChainUrl;
+
+	public String getDefaultTrustAnchor() {
+		return defaultTrustAnchor;
+	}
+
+	public List<ProviderInfo> getCieProviders() {
+		return cieProviders;
+	}
+
+	public List<ProviderInfo> getSpidProviders() {
+		return spidProviders;
+	}
+
+	public Map<String, String> getIdentityProviders(OIDCProfile profile) {
+		Map<String, String> result = new HashMap<>();
+
+		if (OIDCProfile.CIE.equals(profile)) {
+			for (ProviderInfo provider : cieProviders) {
+				result.put(provider.getSubject(), provider.getTrustAnchor());
+			}
+		}
+		else if (OIDCProfile.SPID.equals(profile)) {
+			for (ProviderInfo provider : spidProviders) {
+				result.put(provider.getSubject(), provider.getTrustAnchor());
+			}
+		}
+
+		return result;
+	}
+
+	public List<String> getTrustAnchors() {
+		return trustAnchors;
+	}
+
+	public RelyingParty getRelyingParty() {
+		return relyingParty;
+	}
+
+	public Hosts getHosts() {
+		return hosts;
+	}
+
+	public void setDefaultTrustAnchor(String defaultTrustAnchor) {
+		this.defaultTrustAnchor = defaultTrustAnchor;
+	}
+
+	public void setTrustAnchors(List<String> trustAnchors) {
+		this.trustAnchors = trustAnchors;
+	}
+
+	public JSONObject toJSON() {
+		JSONObject json = new JSONObject();
+
+		json.put("defaultTrustAnchor", defaultTrustAnchor);
+		json.put("trustAnchors", trustAnchors);
+		json.put("relyingParty", relyingParty.toJSON());
+		json.put("spidProviders", new JSONArray(spidProviders));
+		json.put("cieProviders", new JSONArray(cieProviders));
```

Comment:
RP, OP, AS, RS, wallet specific roles, should be taken out from this implementation which scope is the openid federation authority

Reply:
Modified, but these properties were configured from a previous version of a library shared in the last projects (i.e. eudi-wallet-it-openid-credential-issuer), there will be a refactor about it in a next version.